src/libPMacc/include/algorithms/math/floatMath/fmod.tpp

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2016 Alexander Debus
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "math.h"
+#include <cmath>
+
+namespace PMacc
+{
+namespace algorithms
+{
+namespace math
+{
+
+template<>
+struct Fmod<float>
+{
+    typedef float result;
+
+    HDINLINE result operator( )(result x, result y)
+    {
+#if __CUDA_ARCH__
+        return ::fmod( x , y );
```

Comment:
looks like someone is `using std`  ;)

you want http://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html